C++ PickFolder, pass UniquePath by reference

In the C++ version, the PickFolder functions do not pass UniquePathN and UniquePathU8 by reference as the other functions do. This results in a compiler error:

error C2280: 'std::unique_ptr<nfdchar_t,NFD::PathDeleter<nfdchar_t>>::unique_ptr(const std::unique_ptr<nfdchar_t,NFD::PathDeleter<nfdchar_t>> &)': attempting to reference a deleted function

https://github.com/btzy/nativefiledialog-extended/blob/916a0af489144c0a8c8c03a8072cea61ca7fe6f5/src/include/nfd.hpp#L196

https://github.com/btzy/nativefiledialog-extended/blob/916a0af489144c0a8c8c03a8072cea61ca7fe6f5/src/include/nfd.hpp#L233

Changing these to references resolved the issue for me.

--- a/src/include/nfd.hpp
+++ b/src/include/nfd.hpp
@@ -193,7 +193,7 @@ namespace NFD {
         return res;
     }
 
-    inline nfdresult_t PickFolder(UniquePathN outPath, const nfdnchar_t* defaultPath = nullptr) noexcept {
+    inline nfdresult_t PickFolder(UniquePathN& outPath, const nfdnchar_t* defaultPath = nullptr) noexcept {
         nfdnchar_t* out;
         nfdresult_t res = PickFolder(out, defaultPath);
         if (res == NFD_OKAY) {
@@ -230,7 +230,7 @@ namespace NFD {
         return res;
     }
 
-    inline nfdresult_t PickFolder(UniquePathU8 outPath, const nfdu8char_t* defaultPath = nullptr) noexcept {
+    inline nfdresult_t PickFolder(UniquePathU8& outPath, const nfdu8char_t* defaultPath = nullptr) noexcept {
         nfdu8char_t* out;
         nfdresult_t res = PickFolder(out, defaultPath);
         if (res == NFD_OKAY) {

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -5,6 +5,7 @@ set(TEST_LIST
   test_opendialogmultiple.c
   test_opendialogmultiple_cpp.cpp
   test_pickfolder.c
+  test_pickfolder_cpp.cpp
   test_savedialog.c)
   
 foreach (TEST ${TEST_LIST})

--- a/test/test_pickfolder_cpp.cpp
+++ b/test/test_pickfolder_cpp.cpp
@@ -1,0 +1,33 @@
+#include "nfd.hpp"
+
+#include <iostream>
+
+/* this test should compile on all supported platforms */
+/* this demonstrates the thin C++ wrapper */
+
+int main()
+{
+    // initialize NFD
+    NFD::Guard nfdGuard;
+
+    // auto-freeing memory
+    NFD::UniquePath outPath;
+
+    // show the dialog
+    nfdresult_t result = NFD::PickFolder(outPath);
+    if (result == NFD_OKAY)
+    {
+        std::cout << "Success!" << std::endl << outPath.get() << std::endl;
+    }
+    else if (result == NFD_CANCEL)
+    {
+        std::cout << "User pressed cancel." << std::endl;
+    }
+    else
+    {
+        std::cout << "Error: " << NFD::GetError() << std::endl;
+    }
+
+    // NFD::Guard will automatically quit NFD.
+    return 0;
+}

